server/src/main/java/org/elasticsearch/common/util/concurrent/EsThreadPoolExecutor.java

```diff
@@ -29,6 +29,10 @@ public class EsThreadPoolExecutor extends ThreadPoolExecutor {
 
     private static final Logger logger = LogManager.getLogger(EsThreadPoolExecutor.class);
 
+    // noop probe to prevent starvation of work in the work queue due to ForceQueuePolicy
+    // https://github.com/elastic/elasticsearch/issues/124667
+    static final Runnable WORKER_PROBE = () -> {};
```

Comment:
Are you sure this is not reference-equal to `() -> {}` coming from outside? I think this might be a compile-time constant in practice. Suggest using `new Runnable() {...}` to ensure uniqueness.

Reply:
It doesn't look, but agreed, this might change any time. Changing 👍

Reply:
Wow I'd really like to know if that can be the case. It _could_ be, but wouldn't this break things? (unless this is specified somewhere).
Better err on the safe side and use the Runnable, but I'm _really_ curious if this could ever happen.

Reply:
Yeah not sure now, I thought I'd seen this but I tried a simple test and couldn't reproduce. That said, [JLS §15.27.4](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.27.4) does specify:

> - A new object need not be allocated on every evaluation.